scvi/dataset/dataset.py

```diff
@@ -32,6 +32,10 @@ def __init__(self, X, local_means, local_vars, batch_indices, labels,
         self.batch_indices, self.n_batches = arrange_categories(batch_indices)
         self.labels, self.n_labels = arrange_categories(labels)
         self.x_coord, self.y_coord = x_coord, y_coord
+        if x_coord is not None:
+            self.got_spatial = True
+        else:
+            self.got_spatial = False
```

Comment:
Let's get rid of the `got_spatial` variable, and just use `x_coord not is None` if we need it. Better to have fewer variables.